File: docs/benchmarks/suite/fixtures/go-service/internal/service/session_service.go
package service

import (
	"example.com/madar-go-service/internal/audit"
	"example.com/madar-go-service/internal/repository"
)

type SessionService struct {
	repository *repository.SessionRepository
	audit      *audit.LoginAudit
}

func NewSessionService(repository *repository.SessionRepository, audit *audit.LoginAudit) *SessionService {
	return &SessionService{
		repository: repository,
		audit:      audit,
	}
}

func (s *SessionService) CreateLoginSession(userID string, deviceID string) repository.Session {
	session := s.repository.CreateLoginSession(userID, deviceID)
	s.audit.RecordLoginSession(userID, session.ID)
	return session
}

File: docs/benchmarks/suite/fixtures/python-service/app/services/session_service.py
from app.audit import AuditLog
from app.repositories.session_repository import SessionRecord, SessionRepository


class SessionService:
    def __init__(self, repository: SessionRepository, audit_log: AuditLog) -> None:
        self.repository = repository
        self.audit_log = audit_log

    def create_login_session(self, user_id: str, device_id: str) -> SessionRecord:
        session = self.repository.create_login_session(user_id=user_id, device_id=device_id)
        self.audit_log.record_login_session(user_id=user_id, session_id=session.session_id)
        return session


_default_repository = SessionRepository()
_default_audit_log = AuditLog()


def get_session_service() -> SessionService:
    return SessionService(repository=_default_repository, audit_log=_default_audit_log)
